src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -2,40 +2,44 @@
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
 
-import java.io.File;
+import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
 import java.nio.file.StandardOpenOption;
 import java.sql.*;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
 public class MapOldStableIds {
 	private static final Logger logger = LogManager.getLogger();
 	private static final String filename = "reactome_stable_ids.txt";
 
-	public static void execute(String username, String password, String host, String releaseNumber) throws Exception {
+	/**
+	 * This DownloadDirectory module produces a mapping file of current Reactome stable identifiers to old Reactome stable identifiers.
+	 * This stable identifiers denote specific instances in Reactome (Pathway, Reaction, Protein) and can be used to access their pages externally.
+	 * After release '53', Reactome switched its stable identifier format from 'REACT_XXXXX' to 'R-ABC-XXXXXX'. This new format
+	 * contained a bit more information ('ABC' denotes the species) while also mitigating a database corruption issue that sometimes caused
+	 * multiple stable identifiers to be mapped to a single instance.
+	 * @param dba MySQLAdaptor - Connects to release_current relational database.
+	 * @param releaseNumber String - Current release number, used for storing files in release-specific location.
+	 * @throws Exception - Can be an IOException (Moving/Writing issues), SQLException (Connecting/Querying/Interacting with stable_identifiers database),
+	 * ClassNotFoundException (if MySQL driver class isn't found) or a general Exception thrown from the MySQLAdaptor class.
+	 */
```

Comment:
If this method is capable of throwing different Exception subclasses, could the signature and the JavaDocs be changed to throw and document the circumstances of each one separately?